### DISCUSSION TITLE
Delete doesn't seem to work

### DISCUSSION THREAD

**Discussion by geoidesic** [zotonic/zotonic#3923] [Q&A] [answered]
How do I really delete media? These were deleted but they're still there.
<img width="1537" alt="Screenshot 2024-11-04 at 20 42 08" src="https://github.com/user-attachments/assets/ad2d0703-ebfa-402f-b06e-689800701480">

I also tried deleting some pages and other things.. some did become greyed out in this list, others refuse to delete (nothing happens). But even the ones that are greyed out, still how up on the site itself – i.e. on the home page as articles. 

Presumably the greyed out lines indicate deletion... or does it indicate publish status? Either way neither deleting nor unpublishing seems to affect what's shown on the home.tpl. 

Why are items in the media list not also greyed out? How do I actually delete something?

<img width="1506" alt="Screenshot 2024-11-04 at 22 21 48" src="https://github.com/user-attachments/assets/64c7fff4-8d1c-477b-b3c7-72f197f32a11">

I am seeing some errors in the logs which don't look encouraging:

```
ERROR  site=arc2  user_id=1  dispatch=page <0.3447.0> [template_compiler_runtime_internal:include_1/7:246] ▸ text="Included template not found" template=_comments.tpl
srcpos="{<<\"/opt/zotonic/_build/default/lib/arc2/priv/templates/page.article.tpl\">>,25,25}" result=error reason=enoent line=25 at=/opt/zotonic/_build/default/lib/arc2/priv/templates/page.article.tpl
2024-11-04 19:47:30 INFO  site=arc2  user_id=1 <0.1570.0> [mod_logging:check_db_pool_health/2:460] ▸ text="Database pool usage is high. Increase site.db_max_connections" site=arc2 result=warning reason=database_pool in=zotonic_core count=4
2024-11-04 20:40:49 WARNING  site=arc2  user_id=1 <0.20693.0> [mod_admin:event/2:475] ▸ text="Error during bulk delete of resources" result=error reason=eacces in=zotonic_mod_admin id=340  
```

**Comment by mworrell** [comment, marked as answer]
The message `Database pool usage is high. Increase site.db_max_connections`  is just that we were waiting quite long for a new database connection.  Might mean that there aren't enough connections available, or that the database server is just a tad slow with making new connections.

The `Included template not found` is exactly that, a template was not found. I see that template in the module `mod_comment`, is it turned on?

The images are imported content, might be that they are protected.  You can open their edit page and uncheck the 'protected' flag.

<img width="576" alt="Screenshot 2024-11-05 at 10 40 49" src="https://github.com/user-attachments/assets/76487ef9-ebdb-40ab-98de-00cf96c4c9c5">

**Reply by geoidesic** [reply]
Ok yes they were protected. Might be worth having a notification in the UI to that effect if the deletion fails, rather than it just not working. Or at least a flag on the Pages list table to indicate that they are protected – maybe a lock icon.

**Reply by geoidesic** [reply]
BUT... it doesn't work for all of them. E.g. I tried to delete the `Zotonic Introduction Video`... nothing happens. If I click on `View`, it takes me to a blank page in the CMS. With this warning in the logs:
```
2024-11-05 11:21:53 WARNING  site=arc2  user_id=1 <0.6204.0> [mod_admin:event/2:475] ▸ text="Error during bulk delete of resources" result=error reason=eacces in=zotonic_mod_admin I’d=340
```

![Screenshot 2024-11-05 at 11 23 25](https://github.com/user-attachments/assets/594a4b12-34c8-47fa-97fc-27b227b5059b)

**Reply by geoidesic** [reply]
@mworrell  any thoughts on this one?

**Reply by mworrell** [reply]
Can you check if the resource is still there?

(You can see the id in the URL of in the http headers)

**Reply by mworrell** [reply]
The warning is just the bulk delete that couldn't delete resource id 340.  Because of access denied (eacces) error.  Might be the protected flag.

**Reply by geoidesic** [reply]
It exists. Here's the SQL:
```sql
insert into "rsc" ("category_id", "content_group_id", "created", "creator_id", "id", "is_authoritative", "is_dependent", "is_featured", "is_protected", "is_published", "is_unfindable", "language", "modified", "modifier_id", "name", "page_path", "pivot_category_nr", "pivot_city", "pivot_country", "pivot_date_end", "pivot_date_end_month_day", "pivot_date_start", "pivot_date_start_month_day", "pivot_first_name", "pivot_gender", "pivot_geocode", "pivot_geocode_qhash", "pivot_location_lat", "pivot_location_lng", "pivot_postcode", "pivot_rtsv", "pivot_state", "pivot_street", "pivot_surname", "pivot_title", "pivot_tsv", "props", "publication_end", "publication_start", "slug", "uri", "version", "visible_for") values (112, 319, '2024-11-03 11:37:18.501453+00', 1, 340, true, false, false, true, true, false, '{"en"}', '2024-11-04 22:01:08+00', 1, 'media_video', NULL, 12000000, '', '', NULL, NULL, NULL, NULL, '', '', NULL, NULL, NULL, NULL, '', '''zpc110'':2 ''zpc112'':3 ''zpo319'':1', '', '', '', 'zotonic introduction video', '''content'':12 ''default'':11 ''group'':13 ''introduct'':2A,5A ''media'':7B,8B ''video'':3A,6A,9B,10B ''zoton'':1A,4A', '01326a3a83740000000d6d0000000b637573746f6d5f736c756764000566616c73656d0000000f646174655f69735f616c6c5f64617964000566616c73656d0000000c696e7374616c6c65645f62796d00000004617263326d0000001269735f6d61696c696e675f6f70745f6f757464000566616c73656d0000001569735f706167655f706174685f6d756c7469706c6564000566616c73656d0000001369735f776562736974655f726564697265637464000566616c73656d0000000d6d616e616765645f70726f707374000000026d0000000a6f656d6265645f75726c6d0000001968747470733a2f2f76696d656f2e636f6d2f373633303931366d000000057469746c656d0000001a5a6f746f6e696320696e74726f64756374696f6e20766964656f6d000000077072697661637961006d0000000b73656f5f6e6f696e64657864000566616c73656d000000057469746c6568026400057472616e736c000000016802640002656e6d0000001a5a6f746f6e696320696e74726f64756374696f6e20766964656f6a6d0000000a7469746c655f736c756768026400057472616e736c000000016802640002656e6d0000001a7a6f746f6e69632d696e74726f64756374696f6e2d766964656f6a6d000000127472616e736c6174696f6e5f73746174757374000000166d0000000261726d000000006d00000002617a6d000000006d0000000264656d000000006d00000002656e6d000000006d0000000265736d000000006d0000000265746d000000006d0000000266726d000000006d0000000268656d000000006d000000026964640009756e646566696e65646d0000000269746d000000006d000000026e6c6d000000006d00000002706c6d000000006d0000000270746d000000006d0000000570742d62726d000000006d0000000272756d000000006d0000000273766d000000006d0000000274686d000000006d00000002746c6d000000006d0000000274726d000000006d00000002756b6d000000006d000000027a686d000000006d000000077a682d68616e746d000000006d00000002747a6d00000003555443', '9999-08-17 12:00:00+00', '2024-11-03 11:37:00+00', 'zotonic-introduction-video', NULL, 3, 0);
```

**Reply by mworrell** [reply]
And in the admin?   `/admin/edit/340`

I'm wondering about those paging arrows, but then I'm not very well known with those simple blog templates.

**Reply by geoidesic** [reply]
Thanks for that – I was being silly and looking at the front-end, not the admin. I can see it is protected in the admin settings. Once I unprotected it I can delete it.

**Reply by geoidesic** [reply]
BUT... after deleting it, it's still listed, just greyed out. And I can still edit it, although DELETE is no longer an option.
I take it this is a soft-delete. Is there a way to either hard delete, or hide deleted items from the list? It threatens to create a lot of clutter in the admin interface if I have to live with deleted content for ever more. 
![Screenshot 2024-11-05 at 14 55 47](https://github.com/user-attachments/assets/6729ae2b-e51e-4aac-a73f-584db5acf46f)

**Reply by mworrell** [reply]
Weird.  Can you flush the caches?  In /admin/status
Maybe some content is hanging around due to whatever reasons.

If stuff is gone from the db, then it should be gone.
And the admin has always delete rights (unless protected...)

**Reply by mworrell** [reply]
We don't have a soft-delete, if something is deleted then it is gone from the database.
(Though the texts are still in the mod_backup revision tables, if you want to recover them.)

**Reply by geoidesic** [reply]
Ok. I flushed the caches in /admin/status and also in the browser and I can confirm that the rsc record is no longer in the database... however it still appears in the pages list. I did unpublish it before deleting it, I don't know if that makes a difference. Screenshot looks the same now as above.

**Reply by geoidesic** [reply]
I logged out and back into the admin section and it still shows all the deleted things.
![Screenshot 2024-11-05 at 15 06 15](https://github.com/user-attachments/assets/4bf9821a-1d17-42cd-a4ff-8e1e10ea1c29)

**Reply by mworrell** [reply]
And a hard-reload of the page in your browser?

This is very weird.

**Reply by geoidesic** [reply]
Yeah I tried that. I even tried logging in to admin on a different browser which I've never used for Zotonic... same deal, the deleted stuff still shows up

**Reply by geoidesic** [reply]
I think you need to take the docker image out for a spin ;-D

**Reply by mworrell** [reply]
haha, I dislike Docker for its indirectness and slowness.

**Reply by geoidesic** [reply]
Ok I got delete to work. I was doing it wrong. I'm easily confused, which makes me a good canary in the coal mine – if I can do it, anyone can! ;P

**Reply by mworrell** [reply]
Great!   And we already got the lock-icon because of your work.  So progress is made 👍

**Comment by geoidesic** [comment]
> The Included template not found is exactly that, a template was not found. I see that template in the module mod_comment, is it turned on?

No, it's off, so I guess I can ignore that error.

**Comment by mworrell** [comment]
Good idea - I'll add a lock symbol after the title (or somewhere else).

**Comment by mworrell** [comment]
Adding lock icon.

https://github.com/zotonic/zotonic/pull/3925
